### DISCUSSION TITLE
How to add support for db-specific operators like `~`

### DISCUSSION THREAD

**Discussion by csotiriou** [Blazebit/blaze-persistence#1753] [General]
I'm using Quarkus 3.1.2, and I need to add support for the postgres regex operator `~`. The default JPA integration allows me to have raw SQL at some point, which I use for this case. However, I cannot find any mention of this in the documentation.

I tried this:
```
var criteria = cb
        .page(0, 10)
        .whereExpression("nameEn ~ 'foo'");
```
And I get an error about the `~` operand. 

Can I have some pointers on how to add support for the ~ operator with postgres, either by using JPQL Functions?

**Comment by beikov** [comment]
Hey there, we have a section about how to create a custom `JPQLFunction` which can render this SQL for you. See https://persistence.blazebit.com/documentation/1.6/core/manual/en_US/#custom-jpql-functions
